iree/hal/dylib/cts/CMakeLists.txt

```diff
@@ -4,30 +4,32 @@
 # See https://llvm.org/LICENSE.txt for license information.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-if(${IREE_HAL_DRIVER_DYLIB})
-
 iree_hal_cts_test_suite(
   DRIVER_NAME
     dylib
   DRIVER_REGISTRATION_HDR
     "iree/hal/dylib/registration/driver_module.h"
   DRIVER_REGISTRATION_FN
     "iree_hal_dylib_driver_module_register"
+  COMPILER_TARGET_BACKEND
+    "dylib-llvm-aot"
+  EXECUTABLE_FORMAT
+    "embedded-elf-x86_64"  # TODO(scotttodd): IREE_ARCH here... somehow? Or reflect?
```

Comment:
Okay, this does need some code to determine the architecture, or a change in how `executable_spec.executable_format` is set.

Here are logs from testing on Android: https://buildkite.com/iree/iree-android-arm64-v8a/builds/5499#1d16c7da-e08f-49a7-8173-6d6685722080
```
local_executable_cache.c:129: NOT_FOUND;
no executable loader registered for the given executable format 'embedded-elf-x86_64'
```

This needs to match in the same way the runtime and compiler already match:

https://github.com/google/iree/blob/ebf3a6a25a5a86c5a68db0ee325d741c8bff4d26/iree/compiler/Dialect/HAL/Target/LLVM/LLVMAOTTarget.cpp#L615-L643

https://github.com/google/iree/blob/ebf3a6a25a5a86c5a68db0ee325d741c8bff4d26/iree/base/target_platform.h#L58-L98

https://github.com/google/iree/blob/ebf3a6a25a5a86c5a68db0ee325d741c8bff4d26/iree/hal/local/loaders/embedded_library_loader.c#L317-L318

It would be nice if this could just use the format from what `iree-translate` produced without needing to line this all up, but I could also just replicate that logic in CMake using `CMAKE_SYSTEM_PROCESSOR` and other methods. Any suggestions, @benvanik ?

Reply:
Another idea: include `IREE_ARCH` in CMake like `embedded-elf-IREE_ARCH` then rely on the generated C file to expand the nested `IREE_ARCH` macro in

```c++
#define IREE_CTS_EXECUTABLE_FORMAT "embedded-elf-IREE_ARCH"

const char* get_test_executable_format() { return IREE_CTS_EXECUTABLE_FORMAT; }
```

somehow, all while keeping the simple case of no nested macro working.

Reply:
Macro approach seems to work... https://godbolt.org/z/oaaG51YT8. A bit annoying needing to include escaped quotes for the common case though.

Reply:
Phew, tests on Android using cross compilation work now too!